common/src/test/org/apache/hive/common/util/TestDateParser.java

```diff
@@ -86,5 +78,8 @@ public void testInvalidCases() throws Exception {
     checkInvalidCase("0000-00-00");
     checkInvalidCase("2001-13-12");
     checkInvalidCase("2001-11-31");
+    checkInvalidCase("19999-10-11");
+    checkInvalidCase("08-08-2023");
+    checkInvalidCase("2023-08-0800");
```

Comment:
I double-checked 4.0.0 returns `9999-10-11`, `0008-08-20`, and `2023-08-08`. The new behavior looks more reasonable to me.

Reply:
Agree.